src/main/scala/com/databricks/labs/validation/Rule.scala

```diff
@@ -95,11 +94,11 @@ class Rule(
       inputColumn.expr.children.map(_.prettyName).contains("aggregateexpression")
   }
 
-  def cleanseRuleName(ruleName: String): String = {
-    if (ruleName.contains(" ")) logger.warn("Replacing whitespaces in Rule Name with underscores.")
-    val removedWhitespaceRuleName = ruleName.trim.replaceAll(" ", "_")
+  def ruleName: String = {
+    if (_ruleName.contains(" ")) logger.warn("Replacing whitespaces in Rule Name with underscores.")
+    val removedWhitespaceRuleName = _ruleName.trim.replaceAll(" ", "_")
     val specialCharsPattern = "[^a-zA-z0-9_-]+".r
-    if (specialCharsPattern.findAllIn(ruleName).toSeq.nonEmpty) logger.warn("Removing special characters from Rule Name.")
+    if (specialCharsPattern.findAllIn(_ruleName).toSeq.nonEmpty) logger.warn("Removing special characters from Rule Name.")
     removedWhitespaceRuleName.replaceAll("[^a-zA-Z0-9_-]", "")
```

Comment:
I think we should allow underscores and dashes as those don't conflict with SQL (that I know of). Additionally, rather than just dropping the characters, I think we should replace the special chars with `_`. This lets the user know that it was stripped as well and where it was stripped from. You good with that?